### ISSUE TITLE
Question about the second decent of instruction tuning loss

### ISSUE THREAD

**Issue by Forence1999** [allenai/open-instruct#128]
When finetuning on all the instruction following datasets, I find that the training loss suddenly decreases at about the 5.3k-th step, which is quite strange. Also, it seems that it isn't related to the learning rate, because the second reduction disappears when I reduce the training steps to 5k.
This phenomenon does confuse me. Anyone has ideas about why this happens? Thanks a lot!

![image](https://github.com/allenai/open-instruct/assets/50961851/4da55323-ab5a-402a-9a49-2ff0be57aad4)

**Comment by hamishivi** [comment]
Hi, my guess is that this is an epoch boundary? We often observe large drops in training loss at the start of an epoch that look similar to this. It happens because the model starts to memorise the data, although this does not hurt its generalization (somewhat surprisingly).

**Comment by Forence1999** [comment]
Thank you, Sir! Yes, I tried another smaller dataset. The training loss suddenly decreases at each epoch boundary.
